test/unit/controllers/UploadWrittenAuthorisationControllerSpec.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import connectors.{DataCacheConnector, FakeDataCacheConnector}
+import controllers.actions._
+import forms.UploadWrittenAuthorisationFormProvider
+import models.{FileAttachment, NormalMode}
+import navigation.FakeNavigator
+import org.mockito.ArgumentMatchers.{any, refEq}
+import org.mockito.BDDMockito.given
+import org.mockito.Mockito.reset
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.mockito.MockitoSugar
+import pages.UploadWrittenAuthorisationPage
+import play.api.data.Form
+import play.api.libs.Files.TemporaryFile
+import play.api.libs.json.Json
+import play.api.mvc.MultipartFormData.FilePart
+import play.api.mvc.{Call, MultipartFormData}
+import play.api.test.Helpers._
+import service.FileService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.cache.client.CacheMap
+import views.html.uploadWrittenAuthorisation
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class UploadWrittenAuthorisationControllerSpec extends ControllerSpecBase with MockitoSugar with BeforeAndAfterEach {
+
+  def onwardRoute = Call("GET", "/foo")
+
+  private val fileService = mock[FileService]
+  private val cacheConnector = mock[DataCacheConnector]
+
+  override protected def beforeEach(): Unit = {
```

Comment:
I would do this in the `afterEach`

Reply:
Which is the extra benefit ?

Reply:
with a test in the form: 
```
override protected def beforeEach(): Unit = {
    // Before Each
}

"something" should {
    // A
    "do something" in {
        // B
    }
}
```

For whatever reason the code block `A` executes first, then the `beforeEach` then code block `B` so you sometimes get false negatives when using mocks and verifys.